docs/user-guide/logging/overview.md

```diff
@@ -0,0 +1,88 @@
+
+When user performs [`kubectl logs`](/docs/user-guide/kubectl/kubectl_logs), as shown above, request ends up on the kubelet, which in turn reads directly from the log file and returns its content in the response. __Note! Only last rotation is returned, the rest should be extracted manually.__
+
+### System components logs
+
+The information above is only about application containers. There are several system components (for example, `kube-proxy`), which go around this mechanism and write directly to the files in `/var/log` directory in the host filesystem. These files are also rotated daily and based on size, but size retention for them is higher, `100MB` instead of `10MB`.
+
+System components use the [glog](https://godoc.org/github.com/golang/glog) logging library. Conventions for logging severity for those components are described in [docs/devel/logging.md](https://github.com/kubernetes/kubernetes/tree/{{page.githubbranch}}/docs/devel/logging.md).
+
+## Cluster-level logging architectures
+
+### Using node logging agent
+
+![Using node level logging agent](/images/docs/user-guide/logging/logging-with-node-agent.png)
+
+Logging agent is a tool to expose or push logs to a backend. From high level perspective it's a container, which has access to a directory with log files from all containers. This container should be on every node, therefore it should be either manifest pod, shipped with a node, or a daemon set replica.
+
+This is the most common and encouraged approach, because only one logging agent per node is created and it doesn't require any changes in the target application. __Note! Currently, it only works for application's standard output and standard error!__
+
+#### Default solutions
+
+Kubernetes doesn't specify logging agent, but it ships with two default options. Both use fluentd with custom configuration as an agent, both are shipped as manifest pods.
+
+* Stackdriver Logging in Google Cloud, more information [there](/docs/user-guide/logging/sdl)
+
+* Elasticsearch inside the cluster, more information [there](/docs/user-guide/logging/elasticsearch)
+
+### Using side-container with the logging agent
+
+![Using side-container with the logging agent](/images/docs/user-guide/logging/logging-with-sidecar.png)
+
+Alternatively, dedicated logging agent with it's own configuration for each application may be used. One way of doing so is to have a sidecar container in pod specs, containing only the logging agent.
```

Comment:
"You can implement cluster-level logging by including a dedicated logging agent _for each application_ on your cluster. You can include this logging agent as a "sidecar" container in the pod spec for each application; the sidecar container should contain only the logging agent."